src/moscot/problems/cross_modality/_translation.py

```diff
@@ -0,0 +1,217 @@
+from types import MappingProxyType
+from typing import Any, Literal, Mapping, Optional, Sequence, Tuple, Type, Union
+
+from anndata import AnnData
+import numpy as np
+
+from moscot import _constants
+from moscot._docs._docs import d
+from moscot._types import (
+    ArrayLike,
+    Policy_t,
+    ProblemStage_t,
+    QuadInitializer_t,
+    ScaleCost_t,
+    Str_Dict_t,
+)
+from moscot.base.problems.compound_problem import B, CompoundProblem, K
+from moscot.base.problems.problem import OTProblem
+from moscot.problems._utils import handle_cost, handle_joint_attr
+from moscot.problems.cross_modality._mixins import CrossModalityTranslationMixin
+from moscot.utils.subset_policy import DummyPolicy, ExternalStarPolicy
+from moscot.base.output import BaseSolverOutput
+
+__all__ = ["TranslationProblem"]
+
+
+@d.dedent
+class TranslationProblem(CompoundProblem[K, OTProblem], CrossModalityTranslationMixin[K, OTProblem]):
+    """
+    Class for integrating single cell multiomics data, based on :cite:`demetci-scot:22`.
+
+    Parameters
+    ----------
+    adata_src
+        Instance of :class:`anndata.AnnData` containing the source data.
+    adata_tgt
+        Instance of :class:`anndata.AnnData` containing the target data.
+    """
+
+    def __init__(self, adata_src: AnnData, adata_tgt: AnnData, **kwargs: Any):
+        super().__init__(adata_src, **kwargs)
+        self._adata_tgt = adata_tgt
+        self.filtered_vars: Optional[Sequence[str]] = None
+
+    def _create_policy(  # type: ignore[override]
+        self,
+        policy: Literal["external_star"] = "external_star",
+        key: Optional[str] = None,
+        **kwargs: Any,
+    ) -> Union[DummyPolicy, ExternalStarPolicy[K]]:
+        """Private class to create DummyPolicy if no batches are present in the spatial anndata."""
+        del policy
+        if key is None:
+            return DummyPolicy(self.adata, **kwargs)
+        return ExternalStarPolicy(self.adata, key=key, **kwargs)
+    
+    def _create_problem(
+        self,
+        src: K,
+        tgt: K,
+        src_mask: ArrayLike,
+        tgt_mask: ArrayLike,
+        **kwargs: Any,
+    ) -> OTProblem:
+        return self._base_problem_type(
+            adata=self.adata_src,
+            adata_tgt=self.adata_tgt,
+            src_obs_mask=src_mask,
+            tgt_obs_mask=None,
+            src_key=src,
+            tgt_key=tgt,
+            **kwargs,
+        )
+
+    @d.dedent
+    def prepare(
+        self,
+        src_attr: Str_Dict_t,
+        tgt_attr: Str_Dict_t,
+        var_names: Optional[Sequence[Any]] = None,
```

Comment:
add `batch_key`, which optionally divides the source adata into multiple batches (in the case of the ExternalStarPolicy